src/uu/split/src/split.rs

```diff
@@ -701,36 +722,74 @@ fn suffix_from(
     ) {
         (true, _, _, _) => {
             suffix_type = SuffixType::Decimal;
-            let suffix_opt = matches.get_one::<String>(OPT_NUMERIC_SUFFIXES); // if option was specified, but without value - this will return None as there is no default value
-            if suffix_opt.is_some() {
-                (suffix_start, suffix_auto_widening) =
-                    handle_long_suffix_opt(suffix_opt.unwrap(), strategy, false)?;
+            let opt = matches.get_one::<String>(OPT_NUMERIC_SUFFIXES); // if option was specified, but without value - this will return None as there is no default value
+            if opt.is_some() {
+                suffix_start = opt
+                    .unwrap()
+                    .parse::<usize>()
+                    .map_err(|_| SettingsError::SuffixNotParsable(opt.unwrap().to_string()))?;
+                suffix_auto_widening = false;
             }
         }
         (_, true, _, _) => {
             suffix_type = SuffixType::Hexadecimal;
-            let suffix_opt = matches.get_one::<String>(OPT_HEX_SUFFIXES); // if option was specified, but without value - this will return None as there is no default value
-            if suffix_opt.is_some() {
-                (suffix_start, suffix_auto_widening) =
-                    handle_long_suffix_opt(suffix_opt.unwrap(), strategy, true)?;
+            let opt = matches.get_one::<String>(OPT_HEX_SUFFIXES); // if option was specified, but without value - this will return None as there is no default value
+            if opt.is_some() {
+                suffix_start = usize::from_str_radix(opt.unwrap(), 16)
+                    .map_err(|_| SettingsError::SuffixNotParsable(opt.unwrap().to_string()))?;
+                suffix_auto_widening = false;
             }
         }
         (_, _, true, _) => suffix_type = SuffixType::Decimal, // short numeric suffix '-d'
         (_, _, _, true) => suffix_type = SuffixType::Hexadecimal, // short hex suffix '-x'
         _ => suffix_type = SuffixType::Alphabetic, // no numeric/hex suffix, using default alphabetic
     }
 
+    // Get suffix length (could be coming from command line of default value)
     let suffix_length_str = matches.get_one::<String>(OPT_SUFFIX_LENGTH).unwrap(); // safe to unwrap here as there is default value for this option
-    let suffix_length: usize = suffix_length_str
+    let mut suffix_length: usize = suffix_length_str
         .parse()
         .map_err(|_| SettingsError::SuffixNotParsable(suffix_length_str.to_string()))?;
 
-    // Override suffix_auto_widening if suffix length value came from command line
-    // and not from default value
-    if matches.value_source(OPT_SUFFIX_LENGTH) == Some(ValueSource::CommandLine) {
+    // Disable dynamic auto-widening if suffix length was specified in command line with value > 0
+    if matches.value_source(OPT_SUFFIX_LENGTH) == Some(ValueSource::CommandLine)
```

Comment:
so, for this option we need to account for the following scenarios as they all have different outcomes, especially in combination with other options (like `--numeric-suffixes`, `--hex-suffixes` and `--number`):
1) value was not provided in cmd line
2) value was provided in cmd line and it is > 0
3) value was provided in cmd line and it is == 0

and these need to be done at least in two different places.

If we drop default value on this options, we would still need to distinguish between these 3 scenarios in all applicable combinations. It would just be different checks and possibly parsing to `usize` more then once or introducing additional temp variables, since we cannot do all of these checks in one place - due to how it works in combination with other options. 

I will try a few different ways and see if it can be simplified.

Reply:
pushed new version that should address this one as well